src/client/java/teammates/client/scripts/DataMigrationForRubricQuestionsWithWeightsAttached.java

```diff
@@ -0,0 +1,112 @@
+
+    public static void main(String[] args) throws IOException {
+        new DataMigrationForRubricQuestionsWithWeightsAttached().doOperationRemotely();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected boolean isPreview() {
+        return true;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected List<Key<FeedbackQuestion>> getEntities() {
+        return ofy().load().type(FeedbackQuestion.class).filter("questionType", FeedbackQuestionType.RUBRIC).keys().list();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected boolean isMigrationNeeded(Key<FeedbackQuestion> questionKey) {
+        FeedbackQuestion question = ofy().load().key(questionKey).now();
+        FeedbackQuestionAttributes attr = FeedbackQuestionAttributes.valueOf(question);
+        FeedbackRubricQuestionDetails fqd = (FeedbackRubricQuestionDetails) attr.getQuestionDetails();
+
+        try {
+            Field hasAssignedWeights = fqd.getClass().getDeclaredField("hasAssignedWeights");
+            hasAssignedWeights.setAccessible(true);
+            if (!hasAssignedWeights.getBoolean(fqd)) {
+                return false;
+            }
+
+            Field rubricWeightsForEachCell = fqd.getClass().getDeclaredField("rubricWeightsForEachCell");
+            rubricWeightsForEachCell.setAccessible(true);
+            List<List<Double>> newWeights = (List<List<Double>>) rubricWeightsForEachCell.get(fqd);
+
+            // Question contains legacy data if rubricWeightsForEachCell list is empty.
+            return newWeights.isEmpty();
+        } catch (ReflectiveOperationException e) {
+            System.out.println(e.getMessage());
+            return true;
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected void printPreviewInformation(Key<FeedbackQuestion> questionKey) {
+        // nothing to do.
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected void migrate(Key<FeedbackQuestion> questionKey) {
+        ofy().transact(() -> {
+            FeedbackQuestion question = ofy().load().key(questionKey).now();
+            question = updateWeights(question);
+            ofy().save().entity(question).now();
+        });
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected void postAction() {
+        // nothing to do
+    }
+
+    protected FeedbackQuestion updateWeights(FeedbackQuestion question) {
+        FeedbackQuestionAttributes attr = FeedbackQuestionAttributes.valueOf(question);
+        FeedbackRubricQuestionDetails fqd = (FeedbackRubricQuestionDetails) attr.getQuestionDetails();
+        List<List<Double>> weightsForEachCell = fqd.getRubricWeights();
+        try {
+            Field rubricWeightsForEachCell = fqd.getClass().getDeclaredField("rubricWeightsForEachCell");
+            rubricWeightsForEachCell.setAccessible(true);
+            rubricWeightsForEachCell.set(fqd, weightsForEachCell);
```

Comment:
Need to delete the old field.

Reply:
@xpdavid I am not sure what is the best way to delete the old field? Should I make the `rubricWeights` as `transient` in the `FeedbackRubricQuestionDetails` file so that while serialization, (L-105) it will not get included?

Reply:
> first converting the 1D list to a 2D list and then on a second iteration deleting the 1D list

That's a good idea.